app/lib/publisher/backend.dart

```diff
@@ -60,6 +64,104 @@ class PublisherBackend {
     });
   }
 
+  /// Create publisher.
+  Future<api.PublisherInfo> createPublisher(
+    String publisherId,
+    api.CreatePublisherRequest body,
+  ) async {
+    InvalidInputException.checkMatchPattern(
+      publisherId,
+      'publisherId',
+      RegExp(r'^[a-z]+\.[a-z]+$'), // validate we have a top-level domain.
+    );
+    InvalidInputException.checkStringLength(
+      publisherId,
+      'publisherId',
+      maximum: 1024, // Some upper limit for sanity.
+    );
+    InvalidInputException.checkNotNull(body.accessToken, 'accessToken');
+    InvalidInputException.checkStringLength(
+      body.accessToken,
+      'accessToken',
+      minimum: 1,
+      maximum: 4096,
+    );
+
+    // Create client for talking to Webmasters API:
+    // https://developers.google.com/webmaster-tools/search-console-api-original/v3/parameters
+    final client = auth.authenticatedClient(
+      http.Client(),
+      auth.AccessCredentials(
+        auth.AccessToken(
+          'Bearer',
+          body.accessToken,
+          DateTime.now().toUtc().add(Duration(minutes: 20)), // avoid refresh
+        ),
+        null,
+        [wmx.WebmastersApi.WebmastersReadonlyScope],
+      ),
+    );
+    try {
+      final sites = await retry(
+        () => wmx.WebmastersApi(client).sites.list(),
+        maxAttempts: 3,
+        maxDelay: Duration(milliseconds: 500),
+        retryIf: (e) => e is! auth.AccessDeniedException,
+      );
+      // Determine if the user is in fact owner of the domain in question.
+      final isOwner = sites.siteEntry.any(
+        (s) =>
+            s.siteUrl == 'sc-domain:$publisherId' &&
+            s.permissionLevel == 'siteOwner',
+      );
+      if (!isOwner) {
+        throw AuthorizationException.userIsNotDomainOwner(publisherId);
+      }
+    } on auth.AccessDeniedException {
+      throw AuthorizationException.missingSearchConsoleReadAccess();
+    } finally {
+      client.close();
+    }
+
+    // Create the publisher
+    await _db.withTransaction((tx) async {
+      final key = _db.emptyKey.append(Publisher, id: publisherId);
+      final p = (await tx.lookup<Publisher>([key])).single;
+      if (p != null) {
+        // Check that publisher is the same as what we would create.
+        final tenMinAgo =
+            DateTime.now().toUtc().subtract(Duration(minutes: 10));
+        if (p.created.isBefore(tenMinAgo) || // approximate what we would create
+            p.updated.isBefore(tenMinAgo) ||
+            p.contactEmail != authenticatedUser.email ||
+            p.description != '' ||
+            p.websiteUrl != 'https://$publisherId') {
+          throw ConflictException.publisherAlreadyExists(publisherId);
+        }
+        // Avoid creating the same publisher again, this end-point is idempotent
+        // if we just do nothing here.
+        return;
+      }
+
+      // Create publisher
+      tx.queueMutations(inserts: [
+        Publisher()
```

Comment:
Also create the first member with the current user.

Reply:
Nice catch! :)